tests/foreman/api/test_capsulecontent.py

```diff
@@ -36,6 +53,40 @@
 from robottelo.utils.issue_handlers import is_open
 
 
+def _capsule_sync_wait_assert(sat, capsule, start_time=None, timeout=600):
+    """Query sync status of a given capsule after invoking capsule sync, find some expected sync task(s).
+    * The capsule's `last_sync_task` is the final task from active_tasks (by highest :id).
+    * After polling sync task(s), capsule's `last_sync_time` is on or after timestamp.
+    * No remaining active or failed task(s), after polling the found active task(s).
+    param start_time; datetime.datetime: time to check final sync was on or after
+    """
+    # get most recent status first, to avoid missing any async task(s)
+    sync_status = capsule.nailgun_capsule.content_get_sync(synchronous=True, timeout=timeout)
+    active_tasks = sync_status['active_sync_tasks']
+
+    if start_time is None:
+        start_time = datetime.utcnow()
+    # 1s margin of safety for rounding
+    start_time = (
+        (start_time - timedelta(seconds=1)).replace(microsecond=0).strftime('%Y-%m-%d %H:%M:%S UTC')
+    )
+    # we expect some ongoing sync task(s)
+    assert len(active_tasks) > 0
+    for task in active_tasks:
+        sat.api.ForemanTask(id=task['id']).poll(timeout=timeout)
```

Comment:
It looks like when we use `synchronous` with `timeout` on line 64, the polling is really done in nailgun, if I'm not quite mistaken. So we should rather assert `len(active_tasks) == 0` here, I guess.

```suggestion
    # we expect no ongoing sync task(s) since using synchronous content_get_sync
    assert len(active_tasks) == 0
```

Reply:
@vsedmik I don't get that expected behavior. `content_get_sync()` I do not think is the actual sync task(s) running, but just the status? 
It currently runs synchronously (default) and more often there are in-progress sync(s) found when called.

Reply:
@vsedmik It often returns in progress `active_syn_task`s, that are not done, locally none exceeded 60s, in prt none exceed 120s (some bulk syncs take around 90s).
per your comment here and @lpramuk, looks like he removed reference to the `capsule_mixins` method. The original code was to update `wait_for_sync()`, not sure if we should leave in mixins or in this module?

Reply:
@damoore044 I would vote for fixing the `wait_for_sync` in mixin [here](https://github.com/SatelliteQE/robottelo/blob/72ec0864d2a926673244c10712cc8af92b16ac3b/robottelo/host_helpers/capsule_mixins.py#L64), if possible. That way we can use it from other modules too like `capsule_configured.wait_for_sync()`.
The question was if we could leverage the polling in nailgun (as @lpramuk proposed), or rather add the logic around (as pushed previously). Here I was not sure if the polling would work as we need.

Reply:
@vsedmik The original changes from ~2 weeks ago included the fix to the `wait_for_sync` method, I will bring those back in for the next commit.